pyaedt/application/MessageManager.py

```diff
@@ -105,16 +101,23 @@ class AEDTMessageManager(object):
     >>> hfss._messenger.add_error_message("This is a project error message", "Project")
 
     """
-    def __init__(self, parent=None, loadondesktop=True, loadonlog=True):
+    def __init__(self, parent=None):
         self._parent = parent
         if not parent:
-            self.MainModule = sys.modules['__main__']
-            self._desktop = self.MainModule.oDesktop
+            if 'oDesktop' in dir(sys.modules['__main__']):
+                self.MainModule = sys.modules['__main__']
+                self._desktop = self.MainModule.oDesktop
+                self._log_on_desktop = os.getenv('PYAEDT_DESKTOP_LOGS', 'True').lower() in ('true', '1', 't')
+            else:
+                self._log_on_desktop = False
+                self._desktop = None
         else:
             self._desktop = self._parent._desktop
+            self._log_on_desktop = os.getenv('PYAEDT_DESKTOP_LOGS', 'True').lower() in ('true', '1', 't')
+        self._log_on_file = os.getenv('PYAEDT_FILE_LOGS', 'True').lower() in ('true', '1', 't')
+        self._log_on_screen = os.getenv('PYAEDT_SCREEN_LOGS', 'True').lower() in ('true', '1', 't')
```

Comment:
Are we printing to stdout by default then?

Reply:
yes. I was tempted to change the default because actually this modification impacts on logger and aedt_exception_handlers that print out (used in linux mainly)